app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt

```diff
@@ -0,0 +1,346 @@
+    if (context !is IabView) {
+      throw IllegalStateException("Regular buy fragment must be attached to IAB activity")
+    }
+    iabView = context
+  }
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    return inflater.inflate(R.layout.merged_appcoins_layout, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    setHeaderInformation()
+    buy_button.text = setBuyButtonText()
+    cancel_button.text = getString(R.string.back_button)
+    setPaymentInformation()
+    setBonus()
+    setBackListener(view)
+    payment_methods.visibility = VISIBLE
+    mergedAppcoinsPresenter.present()
+  }
+
+  private fun setBuyButtonText(): String {
+    return if (isDonation) getString(R.string.action_donate) else getString(R.string.action_buy)
+  }
+
+  private fun setBonus() {
+    //Build string for both landscape (header) and portrait (radio button) bonus layout
+    appcoins_radio?.bonus_value?.text =
+        getString(R.string.gamification_purchase_header_part_2, bonus)
+    bonus_value?.text = getString(R.string.gamification_purchase_header_part_2, bonus)
+
+    //Set visibility for both landscape (header) and portrait (radio button) bonus layout
+    if (appcoins_radio_button.isChecked) {
+      bonus_layout?.visibility = VISIBLE
+      bonus_msg?.visibility = VISIBLE
+    }
+    appcoins_bonus_layout?.visibility = VISIBLE
+  }
+
+  private fun setHeaderInformation() {
+    if (isDonation) {
+      app_name.text = getString(R.string.item_donation)
+      app_sku_description.text = getString(R.string.item_donation)
+    } else {
+      app_name.text = getApplicationName(appName)
+      app_sku_description.text = productName
+    }
+    try {
+      app_icon.setImageDrawable(context!!.packageManager
+          .getApplicationIcon(appName))
+    } catch (e: PackageManager.NameNotFoundException) {
+      e.printStackTrace()
+    }
+    val formatter = Formatter()
+    val decimalFormat = DecimalFormat("0.00")
+    val appcText = formatter.format(Locale.getDefault(), "%(,.2f", appcAmount)
+        .toString() + " APPC"
+    val fiatText = decimalFormat.format(fiatAmount) + ' ' + currency
+    fiat_price.text = fiatText
+    appc_price.text = appcText
+  }
+
+  private fun setPaymentInformation() {
+    if (appcEnabled) {
+      appcoins_radio.setOnClickListener { appcoins_radio_button.isChecked = true }
+      appcoins_radio_button.setOnCheckedChangeListener { _, checked ->
+        if (checked) paymentSelectionSubject?.onNext(APPC)
+        credits_radio_button.isChecked = !checked
+      }
+      appcoins_radio_button.isEnabled = true
+    } else {
+      appcoins_radio.message.text = getString(R.string.purchase_appcoins_noavailable_body)
+      appcoins_radio.title.setTextColor(resources.getColor(R.color.btn_disable_snd_color))
```

Comment:
`getColor(@ColorRes int id)` is deprecated.
We should use `ContextCompat.getColor(Context context, @ColorRes int id)` instead